x/tokenomics/token_logic_module/global_mint.go

```diff
@@ -249,29 +276,51 @@ func sendRewardsToAccount(
-	globalInflationPerClaim float64,
-) (cosmostypes.Coin, big.Float) {
+	globalInflationPerClaimRat *big.Rat,
+) cosmostypes.Coin {
 	// Determine how much new uPOKT to mint based on global per claim inflation.
-	// TODO_MAINNET: Consider using fixed point arithmetic for deterministic results.
-	settlementAmtFloat := new(big.Float).SetUint64(settlementCoin.Amount.Uint64())
-	newMintAmtFloat := new(big.Float).Mul(settlementAmtFloat, big.NewFloat(globalInflationPerClaim))
-	// DEV_NOTE: If new mint is less than 1 and more than 0, ceil it to 1 so that
-	// we never expect to process a claim with 0 minted tokens.
-	if newMintAmtFloat.Cmp(big.NewFloat(1)) < 0 && newMintAmtFloat.Cmp(big.NewFloat(0)) > 0 {
-		newMintAmtFloat = big.NewFloat(1)
+	settlementAmtRat := new(big.Rat).SetInt(settlementCoin.Amount.BigInt())
+	newMintAmtRat := new(big.Rat).Mul(settlementAmtRat, globalInflationPerClaimRat)
+	// Always ceil the new mint amount.
+	// DEV_NOTE: Since settlementCoin is never zero and the mint amount is ceiled,
+	// mintAmtCoin will always be greater than zero.
+	newMintRem := new(big.Int)
+	newMintAmt, newMintRem := new(big.Int).QuoRem(newMintAmtRat.Num(), newMintAmtRat.Denom(), newMintRem)
+	// If there is a remainder, add one to the mint amount to ceil the value.
+	if newMintRem.Cmp(big.NewInt(0)) > 0 {
+		newMintAmt.Add(newMintAmt, big.NewInt(1))
 	}
-	newMintAmtInt, _ := newMintAmtFloat.Int64()
-	mintAmtCoin := cosmostypes.NewCoin(volatile.DenomuPOKT, math.NewInt(newMintAmtInt))
-	return mintAmtCoin, *newMintAmtFloat
+	mintAmtCoin := cosmostypes.NewCoin(volatile.DenomuPOKT, math.NewIntFromBigInt(newMintAmt))
+	return mintAmtCoin
 }
 
-// calculateAllocationAmount does big float arithmetic to determine the absolute
-// amount from amountFloat based on the allocation percentage provided.
-// TODO_MAINNET(@bryanchriswhite): Measure and limit the precision loss here.
+// calculateAllocationAmount does big.Rat arithmetic to determine the absolute
+// amount from amountInt based on the allocation percentage provided.
 func calculateAllocationAmount(
-	amountFloat *big.Float,
-	allocationPercentage float64,
-) int64 {
-	coinsToAccAmt, _ := big.NewFloat(0).Mul(amountFloat, big.NewFloat(allocationPercentage)).Int64()
-	return coinsToAccAmt
+	amountInt math.Int,
+	allocationPercentageRat *big.Rat,
+) math.Int {
+	amountRat := new(big.Rat).SetInt(amountInt.BigInt())
+
+	allocationRat := new(big.Rat).Mul(amountRat, allocationPercentageRat)
+	allocationAmtInt := new(big.Int).Quo(allocationRat.Num(), allocationRat.Denom())
+
+	return math.NewIntFromBigInt(allocationAmtInt)
+}
+
+// Float64ToRat converts a float64 to a big.Rat for precise decimal arithmetic.
+// TODO_CONSIDERATION: Future versions of CosmosSDK will deprecate float64 values
+// with zero copy encoding of scalar values.
+// We should consider switching to string representations for tokenomics allocation percentages.
+// NB: It is publicly exposed to be used in the tests.
```

Comment:
```suggestion
// Float64ToRat converts a float64 to a big.Rat for precise decimal arithmetic.
// TODO_FUTURE: Consider switching to string representations for tokenomic % allocations
// since CosmosSDK will deprecate float64 values with zero copy encoding of scalar values.
// Ref: https://docs.cosmos.network/main/build/rfc/rfc-002-zero-copy-encoding
// NB: It is publicly exposed to be used in the tests.
```